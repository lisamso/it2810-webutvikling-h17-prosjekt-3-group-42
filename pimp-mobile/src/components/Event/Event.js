import React, { Component } from 'react';
import { Text, View } from "react-native";
import PropTypes from 'prop-types';
import { styles } from '../../styles.js';

class Event extends Component {

  render() {

    return(
      <View style={styles.event}>
        <Text style={[styles.title, styles.emerald]}>{this.props.title}</Text>
        <Text style={[styles.text, styles.emerald]}>{this.props.text}</Text>
        <Text style={[styles.text, styles.emerald]}>On {this.props.date}</Text>

      </View>
    )
  }
}

Event.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
  date: PropTypes.string
}

export default Event