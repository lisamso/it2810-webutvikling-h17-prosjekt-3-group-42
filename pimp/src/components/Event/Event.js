import React, { Component } from 'react';
import PropTypes from 'prop-types'; // ES6

class Event extends Component {
  constructor(props){
    super(props);
    this.handleDelete = this.handleDelete.bind(this);
  }

  handleDelete() {
    console.log("Delete!");
    let LS = JSON.parse(
      localStorage.getItem('events'));
    if(LS) {
      let searchTerm = this.props.id,
        index = -1;
      for(let i = 0, len = LS.length; i < len; i++) {
        if (LS[i].id === searchTerm) {
          index = i;
          break;
        }
      }
      LS.splice(index, 1);
      localStorage.setItem('events',
        JSON.stringify(LS));
      this.props.update();
    }
  }

  render() {
    return (
      <div className='event'>
        <h2>{this.props.title}</h2>
        <p>{`> ${this.props.text}`}</p>
        <div id='event-at'>
          <hr/>
          <p>The event is on {this.props.eventAt}</p>
        </div>
        <button className="someButton" onClick={this.handleDelete}>Delete</button>
      </div>
    )
  }
}

Event.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  text: PropTypes.string,
  eventAt: PropTypes.string,
  update: PropTypes.func
};

export default Event;
