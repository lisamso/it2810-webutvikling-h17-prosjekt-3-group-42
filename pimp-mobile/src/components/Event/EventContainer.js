import React, {Component} from 'react'
import { Text, TextInput, TouchableNativeFeedback, View} from "react-native";
import Modal from 'react-native-modal';
import Event from "./Event";
import { styles } from '../../styles.js';
import DatePicker from "react-native-datepicker";

class EventContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      events: [],
      modalVisible: false,
      title: '',
      text: '',
      date: '',
      count: 0
    };
  }

  componentDidMount() {
  }

  toggleModal() {
    this.setState((prevState) => {
      return {modalVisible: !prevState.modalVisible};
    });
  }

  submitEvent() {
    if (this.state.title !== '') {
      this.newEvent(this.state.title, this.state.text, this.state.date);
    }
    this.setState({title: '', text: ''});
    this.toggleModal()
  }

  removeEvent(id) {
    let arr = this.state.events;
    let index = -1;
    for (let i = 0; i < arr.length; i++) {
      if (arr[i].props.id === id) {
        index = i;
      }
    }
    if (index !== -1) {
      arr.splice(index, 1);
    }
    this.setState({events: arr});
  }

  newEvent(title='',text='', date){
    let event =
      <Event
        id={this.state.count}
        title={title}
        text={text}
        date={date}
      />;
    this.setState((prevState) => {
      return {count: prevState.count + 1}
    });
    let arr = this.state.events;
    arr.push(event);
    this.setState({ events: arr })
  }

  //Not used. Expo kept crashing when I tried implementing sorting based on date of the event
  //Key was the unix time of the event
  /*  sortEvents(){
    let events = [...this.state.events];
    events.sort((a,b) => a.key - b.key);
    this.setState({events: events});
  }*/

  allEvents() {
    return(
      this.state.events.map((n,idx) =>
        <View style={styles.row_box} key={idx}>
          <View style={[styles.container, styles.emerald_border]}>
            {n}
          </View>
          <TouchableNativeFeedback
            onPress={() => { this.removeEvent(n.props.id)}}
          >
            <View style={styles.delete_container}>
              <View style={[styles.delete_border, styles.emerald_border]}>
                <Text style={[styles.delete_text, styles.emerald]}>x</Text>
              </View>
            </View>
          </TouchableNativeFeedback>
        </View>
      ));
  }

  render(){
    return(
      <View style={styles.box}>
        <View style={[styles.box_header, styles.emerald_border]}>
          <Text style={[styles.title, styles.emerald]}>Your Events</Text>
        </View>
        <View style={[styles.box_body, styles.emerald_border]}>
          {this.allEvents()}
        </View>
        <TouchableNativeFeedback
          onPress={() => { this.toggleModal()}}>
          <View style={[styles.box_button, styles.emerald_background]}>
            <Text style={styles.wet_asphalt}>
              New event
            </Text>
          </View>
        </TouchableNativeFeedback>

        <Modal
          animationIn="slideInLeft"
          animationOut="slideOutRight"
          transparent={true}
          isVisible={this.state.modalVisible}
          backDropColor="white"
          onBackdropPress={() => {this.toggleModal()}}
          onBackButtonPress={() => {this.toggleModal()}}
        >
          <View style={styles.box}>
            <View style={[styles.box_header, styles.emerald_border]}>
              <Text style={[styles.title, styles.emerald]}>New event</Text>
            </View>
            <View style={[styles.box_body, styles.emerald_border]}>
              <TextInput
                style={{height: 60, color: '#2ecc71'}}
                placeholder="Title"
                onChangeText={(text) => this.setState({title: text})}
              />
              <TextInput
                style={{height: 60, color: '#2ecc71'}}
                placeholder="Text"
                onChangeText={(text) => this.setState({text: text})}
              />
              <DatePicker
                style={{width: 200}}
                date={this.state.date}
                mode="datetime"
                placeholder="select date"
                format="DD-MM HH:mm"
                minuteInterval={10}
                confirmBtnText="Confirm"
                cancelBtnText="Cancel"
                customStyles={{
                  dateIcon: {
                    position: 'absolute',
                    left: 0,
                    top: 4,
                    marginLeft: 0
                  },
                  dateInput: {
                    marginLeft: 36
                  },
                  dateText: {
                    color: '#2ecc71'
                  }
                }}
                onDateChange={(date) => {this.setState({date: date})}}
              />
            </View>
            <TouchableNativeFeedback
              onPress={() => {this.submitEvent()}}>
              <View style={[styles.box_button, styles.emerald_background]}>
                <Text style={styles.wet_asphalt}>
                  Confirm
                </Text>
              </View>
            </TouchableNativeFeedback>
          </View>
        </Modal>
      </View>

    )
  }
}

EventContainer.propTypes = {

}


export default EventContainer

