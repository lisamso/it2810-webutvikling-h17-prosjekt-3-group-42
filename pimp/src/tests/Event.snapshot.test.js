import Event from '../components/Event/Event';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });



test('Event component should render as expected', () =>{
  const event = {id: '1', title: "SomeEvent", text: "AtSomePlace", eventAt: "Here"};
  const component = shallow(<Event />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

describe('Event button handles click', () =>{

  it('Delete button clicked', () =>{
    const event = {id: '1', title: "SomeEvent", text: "AtSomePlace", eventAt: "Here"};
    const component = shallow(<Event event={event} />);
    const b = component.find('button');
    b.simulate('click');
    const tree = toJson(component);
    console.log(tree);
  });

});
