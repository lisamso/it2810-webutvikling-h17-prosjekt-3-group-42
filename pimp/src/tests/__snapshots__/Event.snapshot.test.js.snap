// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event button handles click Delete button clicked 1`] = `
<div
  className="event"
>
  <h2 />
  <p>
    &gt; undefined
  </p>
  <div
    id="event-at"
  >
    <hr />
    <p>
      The event is on 
    </p>
  </div>
  <button
    className="someButton"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;

exports[`Event component should render as expected 1`] = `
<div
  className="event"
>
  <h2 />
  <p>
    &gt; undefined
  </p>
  <div
    id="event-at"
  >
    <hr />
    <p>
      The event is on 
    </p>
  </div>
  <button
    className="someButton"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
